### ISSUE TITLE
RepoDigests 'list index out of range' error

### ISSUE THREAD

**Issue by kashizui** [codalab/codalab-worksheets#702]
This is because `RepoDigests` is an empty list for images that were pulled before Docker repository v2 was released.

The offending code in `worker.docker_client`:
```
    def get_image_repo_digest(self, request_docker_image):
        logger.debug('Fetching Docker image metadata for %s', request_docker_image)
        with closing(self._create_connection()) as conn:
            conn.request('GET', '/images/%s/json' % request_docker_image)
            create_image_response = conn.getresponse()
            if create_image_response.status != 200:
                raise DockerException(create_image_response.read())
            contents = json.loads(create_image_response.read())
        return contents.get('RepoDigests', [''])[0]
```

@fabeschan Let's make this more robust. This will be a problem for anyone upgrading their CodaLab instance/worker. Maybe at the minimum, some logic to either fall back to empty string `''` or the image tag if `RepoDigests` is an empty list?

**Comment by fabeschan** [comment]
Hmm that is weird. Looks like I've already attempted to implement some sort of fall back; doesn't `contents.get('RepoDigests', [''])` return a list consisting of one element, an empty string if RepoDigest does not exist? Perhaps I missed something

**Comment by kashizui** [comment]
@fabeschan The problem is that `RepoDigests` DOES exist in the JSON, and it's an empty list.

**Comment by kashizui** [comment]
So it doesn't fallback to your default `['']`, and instead tries to get the first element of an empty list.

**Comment by fabeschan** [comment]
Alright, that makes sense

**Comment by fabeschan** [comment]
Percy encountered this issue again. Details:

Everything on master.  Ubuntu 16.04 on my laptop.
Tried running: cl run date --request-docker-image tensorflow/tensorflow

Worker output:
2017-04-06 10:42:34,383 Starting run with UUID 0xbbb67720bbdf47f2b74318fca0c9d4c5
2017-04-06 10:42:34,471 Starting Docker container for UUID 0xbbb67720bbdf47f2b74318fca0c9d4c5 with command date, container ID 1aa266b2cb84bc04a36307954feeec3ae955037aef866e10f0dfdecb02e25148
2017-04-06 10:42:34,653 Fetching Docker image metadata for tensorflow/tensorflow
2017-04-06 10:42:34,655 Finished run with UUID 0xbbb67720bbdf47f2b74318fca0c9d4c5, exitcode None, failure_message list index out of range
2017-04-06 10:42:34,655 Deleting container with ID 1aa266b2cb84bc04a36307954feeec3ae955037aef866e10f0dfdecb02e25148
2017-04-06 10:42:34,804 Uploading results for run with UUID 0xbbb67720bbdf47f2b74318fca0c9d4c5
2017-04-06 10:42:34,904 Finalizing run with UUID 0xbbb67720bbdf47f2b74318fca0c9d4c5

% docker -v
Docker version 1.11.2, build b9f10c9

% docker inspect tensorflow/tensorflow
[
   {
       "Id": "sha256:0bb45d441a4b22efbb3883a1a947e7e6ab4a737ca39455a8c4e6dc3e08367e40",
       "RepoTags": [
           "tensorflow/tensorflow:latest"
       ],
       "RepoDigests": [],
...

**Comment by kashizui** [comment]
I think I found the culprit:
```
        try:
            return image['RepoDigests'][0]
        except KeyError, IndexError:
            return ''
```
It's missing parentheses for catching multiple exception types:
```
        try:
            return image['RepoDigests'][0]
        except (KeyError, IndexError):
            return ''
```
http://stackoverflow.com/questions/6470428/catch-multiple-exceptions-in-one-line-except-block

**Comment by kashizui** [comment]
I can include the fix in #676
